python/nav/ipdevpoll/shadows/__init__.py

```diff
@@ -260,6 +264,51 @@ def _fix_binary_garbage(self):
                 setattr(self, attr, repr(value))
         self.clear_cached_objects()
 
+    def _get_new_device_event(self):
+        if not self.is_new:
+            return
+        device = self.get_existing_model()
+        if not device.serial:
+            return
+        # Check for chassis before module
+        # A device can be a module and a chassis, but chassis is prioritized
+        entity = manage.NetboxEntity.objects.filter(device=device).first()
```

Comment:
This could be replaced by the potentially more succinct
```suggestion
        entity = device.netboxentity_set.first()
```